sdcm/logcollector.py

```diff
@@ -1002,6 +1004,85 @@ def create_archive_per_file_and_upload(self) -> list[str]:
         return s3_links
 
 
+class KubernetesAPIServerLogCollector(SCTLogCollector):
+    """Gather K8S API server logs."""
+    log_entities = [
+        FileLog(name='kube-apiserver-audit.log', search_locally=True),
+    ]
+    cluster_log_type = "kubernetes-apiserver"
+    cluster_dir_prefix = "k8s-"
+    collect_timeout = 600
+
+    def collect_logs(self, local_search_path: Optional[str] = None) -> list[str]:
+        audit_file_name = 'kube-apiserver-audit.log'
+        for ent in self.log_entities:
+            local_path = ent.collect(None, self.local_dir, None, local_search_path=local_search_path)
+            if ent.name == audit_file_name:
+                LOGGER.info("DEBUG: 'self.generate_apiserver_call_stats_file' method has been called")
+                self.generate_apiserver_call_stats_file(local_path + f"/{audit_file_name}")
+        if not os.listdir(self.local_dir):
+            LOGGER.warning('No any local files')
+            LOGGER.info('Searching on builders')
+            builders = get_builder_by_test_id(self.test_id)
```

Comment:
aren't we using sct runners on all k8s pipelines ?, why do we need to search the all the builders ?

Reply:
We have perf jobs which run on builders. And log collector is not tied to local K8S.

Reply:
The pref tests are now using sct-runner (it was merged ~20 days ago, in https://github.com/scylladb/scylla-cluster-tests/commit/fe03b8762c610dbf6eab1ff17e840b02b3e04181)